Bottom: 89e6006df14eb98451d4899acc0456fdf5e16544
Top:    cc9756bbdca9639c9e616385a447d2628b32043d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 17:06:57 +0000

Refresh of play-with-quad-edge-data

---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index f6ebcdd..3e94720 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -3,6 +3,8 @@
 #include <assert.h>
 #include <stdbool.h>
 
+#include <glib.h>
+
 #include "quad.h"
 #include "vertex3d.h"
 #include "face3d.h"
@@ -67,7 +69,7 @@ make_object3d ()
   static int object3d_count = 0;
   object3d *object;
 
-  object = malloc (sizeof (object3d));
+  object = g_new0 (object3d, 1);
   object->id = object3d_count++;
 
   return object;
@@ -77,7 +79,7 @@ void
 destroy_object3d (object3d *object)
 {
   /* XXX: LEAK GEOMETERY AND TOPOLOGY */
-  free (object);
+  g_free (object);
 }
 
 #define XOFFSET 50
